src/components/measurements/index.tsx

```diff
@@ -254,9 +256,25 @@ const MeasurementsPlot = ({height, width, showLegend, setPanelTitle}) => {
     setHoverData(newHoverData);
   }, [fields, colorings, colorBy]);
 
+  /**
+   * Ref to save previous non-measurements coloring for toggling back to previous
+   * coloring when clicking on the same measurements grouping twice.
+   * Uses the default color by if the color is a measurements color on first
+   * load, i.e. the color is set by the URL param `c=m-<grouping>`
+   */
+  const prevNonMeasurementColorBy: MutableRefObject<string> = useRef(isMeasurementColorBy(colorBy) ? defaultColorBy : colorBy);
+  useEffect(() => {
+    if (!isMeasurementColorBy(colorBy)) {
+      prevNonMeasurementColorBy.current = colorBy;
+    }
+  }, [colorBy]);
+
   const handleClickOnGrouping = useCallback((grouping: string): void => {
     if (grouping !== colorGrouping || !isMeasurementColorBy(colorBy)) {
       dispatch(applyMeasurementsColorBy(grouping));
+    } else if (grouping === colorGrouping && isMeasurementColorBy(colorBy)) {
+      // Clicking on the same grouping twice will toggle back to the previous non-measurements coloring
+      dispatch(changeColorBy(prevNonMeasurementColorBy.current));
```

Comment:
[Comment "Click on measurements grouping to toggle back to previous coloring"]

This'll leave the `_hasMeasurementsColor` filter enabled. Just a heads up - not a comment on whether that's the desired behaviour or not.

Reply:
Thanks for bringing this up! I left this in as a "feature", but not sure if others noticed.  

I thought it'd be useful to see the filtered tips in the other coloring option as you quickly toggle back and forth by clicking on the measurements grouping. I'm leaving it as-is unless I hear otherwise in the future.